### PR TITLE
Add mhv route guard to after visit summary

### PR DESCRIPTION
**Pull request by jzucadi**
The purpose of this PR is to implement a route guard for the tool pages under the /my-health/ umbrella. The route guard checks for the presence of mhvAccountState having a value of ‘OK’ or ‘MULTIPLE’ to indicate whether a user has an MHV account. When a user does not have an MHV account, the property mhvAccountState will have a value of ‘NONE’. When a user has an MHV account, they are permitted to access the various tools under /my-health/. When a user does not have an MHV account, the route guard will redirect the user to the /my-health/ landing page where an alert notification will be shown explaining that they do not have an MHV account and need to register. This route guard will prevent the need to have this alert notification on each of the /my-health/ too pages. 

When a user does not have an MHV account, they are redirected to the /my-health/ landing page and will see the following:

Screenshot 2024-04-24 at 1 37 20 PM

Ticket:
https://app.zenhub.com/workspaces/mhv-on-vagov-landing-page-62619a987d74510018ecc546/issues/gh/department-of-veterans-affairs/va.gov-team/77237

## Testing done

To test this, mock user data must be used and toggled between values of 'NONE' = no mhv account and 'OK' or 'MULTIPLE' = has an mhv Account.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by asg5704 — DISMISSED** [review]
The `IdentityNotVerified` component looks good. Also wanted to note that I believe this component was updated recently by your team to add a disable analytics prop to the `va-alert` so just wanted to call that out here as well.

**Review by pjhill — DISMISSED** [review]

File: src/platform/mhv/hooks/useMyHealthAccessGuard.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { useSelector } from 'react-redux';
+
+/**
+ * Route guard hook that will redirect the user to the /my-health landing page if mhvAccountState is 'NONE'
+ */
+
+export const useMyHealthAccessGuard = () => {
```
Comment (carlosfelix2) [inline src/platform/mhv/hooks/useMyHealthAccessGuard.jsx:7]: Perhaps change the name of this function. The name kind of implies to me that the return would be boolean.

File: src/applications/avs/routes.jsx [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 
-const ErrorBoundaryWrapper = props => (
-  <ErrorBoundary>
-    <Avs {...props} />
-  </ErrorBoundary>
-);
+const ErrorBoundaryWrapper = props => {
+  const accessGuard = useMyHealthAccessGuard();
+
+  if (accessGuard) {
```
Comment (carlosfelix2) [inline src/applications/avs/routes.jsx:12]: I don't think this is necessary. The function will redirect automatically. Just call the function without the if

File: src/platform/user/authorization/components/IdentityNotVerified.jsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ const VerifyIdentityInfo = () => (
   >
     <a
       href="https://www.va.gov/resources/verifying-your-identity-on-vagov/"
-      clasName="vads-u-padding-bottom--2"
+      className="vads-u-padding-bottom--2"
```
Comment (carlosfelix2) [inline src/platform/user/authorization/components/IdentityNotVerified.jsx:25]: Ouch!! Nice catch.

File: src/platform/mhv/hooks/useMyHealthAccessGuard.jsx (hunk 1 above)
Comment (jzucadi) [reply src/platform/mhv/hooks/useMyHealthAccessGuard.jsx:7]: The "use" prefix is to indicate that this is a custom hook.

**Comment by asg5704** [comment]
@jzucadi do you need a review on this? I don't see any Identity code 👀

**Review by asg5704 — DISMISSED** [review]

**Comment by jzucadi** [comment]
@pjhill can you please re-review? small changes were made and caused your previous review to be dismissed

**Review by pjhill — APPROVED** [review]

**Pull request merged**